sources/packages/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -70,17 +78,38 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
       .where("educationProgram.id = :programId", { programId })
       .andWhere("institutionLocation.id = :locationId", { locationId });
     if (offeringTypes) {
-      offeringsQuery = offeringsQuery.andWhere(
-        "offerings.offeringType in (:...offeringTypes)",
-        {
-          offeringTypes,
-        },
+      offeringsQuery.andWhere("offerings.offeringType in (:...offeringTypes)", {
+        offeringTypes,
+      });
+    }
+    // search offering name
+    if (paginationOptions?.searchName) {
+      offeringsQuery.andWhere("offerings.name Ilike :searchName", {
+        searchName: `%${paginationOptions.searchName}%`,
+      });
+    }
+    // sorting
+    if (paginationOptions?.sortField && paginationOptions?.sortOrder) {
+      offeringsQuery.orderBy(
+        databaseFieldOfOfferingDataTable(paginationOptions.sortField),
+        paginationOptions.sortOrder,
+      );
+    } else {
+      // default sort and order
+      offeringsQuery.orderBy(
+        databaseFieldOfOfferingDataTable(DEFAULT_SORT_FIELD),
+        FieldSortOrder.ASC,
       );
     }
+    // pagination
+    offeringsQuery
+      .take(paginationOptions.pageLimit)
+      .skip(paginationOptions.page * paginationOptions.pageLimit);
 
-    const queryResult = await offeringsQuery.getMany();
+    // result
+    const queryResult = await offeringsQuery.getManyAndCount();
 
-    return queryResult.map((educationProgramOffering) => {
+    const offerings = queryResult[0].map((educationProgramOffering) => {
```

Comment:
Can you please destructor the array in line 110, something like below to avoid the [0] access?
```javascript
// result
const [records, count] = await offeringsQuery.getManyAndCount();
```